test/unit/math/prim/mat/fun/to_matrix_test.cpp

```diff
@@ -6,15 +6,39 @@
 
 using stan::math::to_matrix;
 
+template <typename T, int R, int C>
+inline Eigen::Matrix<T, R, C>
+row_major_to_column_major(const Eigen::Matrix<T, R, C>& x) {
+  int rows = x.rows();
+  int cols = x.cols();
+  Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+    result(rows, cols);
+  for (int i=0, ij=0; i < rows; i++)
```

Comment:
Looks like you still missed the `i = 0` and `ij = 0` case in lines 16 and 17 (and presumably elsewhere).